README.md

```diff
@@ -1,20 +1,32 @@
 
 If you are looking for something that excels at executing a large queue of tasks in parallel as efficiently as possible, you should take a look at the [parallel](https://github.com/grosser/parallel) project.
 
-## Methods:
+# Install
+```gem install in-parallel```
+
+# Usage
+```include InParallel``` to use as a mix-in
+
+The methods below allow you to fork processes to execute multiple methods or blocks within an enumerable in parallel.  They all have this common behavior:
+1. STDOUT is captured for each forked process and logged all at once when the process completes or is terminated.
+1. By default execution of processes in parallel will wait until execution of all processes are complete before continuing (with the exception of run_in_background).
+    1. You can specify the parameter kill_all_on_error=true if you want to immediately exit all forked processes when an error executing any of the forked processes occurs.
+1. When the forked process raises an exception or exits with a non zero exit code, an exception will be raised in the main process.
+1. Terminating the main process with 'ctrl-c' or killing the process in some other way will immediately cause all forked processes to be killed and log their STDOUT up to that point.
+1. If the result of the method or block can be marshalled, it will be returned as though it was executed within the same process.  If the result cannot be marshalled a warning is produced and the return value will be nil.
+    1. NOTE: results of methods within run_in_parallel can be assigned to instance or class variables, but not local variables.  See examples below.
+1. Will timeout (stop execution and raise an exception) based on a global timeout value, or timeout parameter.
 
+## Methods
 ### run_in_parallel(timeout=nil, kill_all_on_error = false, &block)
 1. Each method in a block will be executed in parallel (unless the method is defined in Kernel or BaseObject).
     1. Any methods further down the stack won't be affected, only the ones directly within the block.
-2. You can assign return values to instance variables and it 'just works'.
-3. Log STDOUT and STDERR chunked per process to the console so that it is easy to see what happened in which process.
-4. Waits for each process in realtime and logs immediately upon completion of each process
-5. If an exception is raised by a child process, it will optionally (kill_all_on_error) be re-raised in the primary process and kill all other still running child processes. The default will wait for all processes to complete execution before re-raising any unhandled exception from the child processes.
-6. Times out by default at 30 minutes. Timeout default can be changed with InParallel::InParallelExecutor.parallel_default_timeout=X, or you can set the timeout param when calling the method
+1. Waits for each process in realtime and logs immediately upon completion of each process
```

Comment:
Yeah, that's just markdown's way of telling it to be a numbered list.  I
think I'm inconsistent and don't do the same in the other number'd lists,
so I'll update them all to be consistent.

On Wed, Jul 20, 2016 at 9:35 AM, Erick Banks <notifications@github.com>
wrote:

> In README.md
> <https://github.com/puppetlabs/in-parallel/pull/9#discussion_r71560238>:
>
> >  ### run_in_parallel(timeout=nil, kill_all_on_error = false, &block)
> >  1. Each method in a block will be executed in parallel (unless the method is defined in Kernel or BaseObject).
> >      1. Any methods further down the stack won't be affected, only the ones directly within the block.
> > -2. You can assign return values to instance variables and it 'just works'.
> > -3. Log STDOUT and STDERR chunked per process to the console so that it is easy to see what happened in which process.
> > -4. Waits for each process in realtime and logs immediately upon completion of each process
> > -5. If an exception is raised by a child process, it will optionally (kill_all_on_error) be re-raised in the primary process and kill all other still running child processes. The default will wait for all processes to complete execution before re-raising any unhandled exception from the child processes.
> > -6. Times out by default at 30 minutes. Timeout default can be changed with InParallel::InParallelExecutor.parallel_default_timeout=X, or you can set the timeout param when calling the method
> > +1. Waits for each process in realtime and logs immediately upon completion of each process
>
> Each of the points here are labeled "1." From line 16 to here... that's a
> lot of "1."s.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/puppetlabs/in-parallel/pull/9/files/c68edb2d21bafdcdd8d80326b708536e12b2ba91#r71560238>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEZD5SfmH-YHbHm13x7I72PijM6jqFHcks5qXk5pgaJpZM4JQ9B7>
> .
>